docs/proposed/psa-driver-interface.md

```diff
@@ -339,11 +339,32 @@ The format of a key for transparent drivers is the same as in applications. Refe
 
 Transparent drivers may provide the following key management entry points:
 
-* `"generate_key"`: called by `psa_generate_key()`, only when generating a key pair (key such that `PSA_KEY_TYPE_IS_ASYMMETRIC` is true).
-* `"key_derivation_output_key"`: called by `psa_key_derivation_output_key()`, only when deriving a key pair (key such that `PSA_KEY_TYPE_IS_ASYMMETRIC` is true).
+* [`"validate_key"`](#key-validation-with-transparent-drivers): called by `psa_import_key()`, only when importing a key pair or a public key (key such that `PSA_KEY_TYPE_IS_ASYMMETRIC` is true).
+* `"generate_key"`: called by `psa_generate_key()`, only when generating a key pair (key such that `PSA_KEY_TYPE_IS_KEY_PAIR` is true).
+* `"key_derivation_output_key"`: called by `psa_key_derivation_output_key()`, only when deriving a key pair (key such that `PSA_KEY_TYPE_IS_KEY_PAIR` is true).
 * `"export_public_key"`: called by the core to obtain the public key of a key pair. The core may call this function at any time to obtain the public key, which can be for `psa_export_public_key()` but also at other times, including during a cryptographic operation that requires the public key such as a call to `psa_verify_message()` on a key pair object.
 
-Transparent drivers are not involved when importing, exporting, copying or destroying keys, or when generating or deriving symmetric keys.
+Transparent drivers are not involved when exporting, copying or destroying keys, or when importing, generating or deriving symmetric keys.
```

Comment:
Because symmetric keys are arbitrary byte strings (of the length given as the key size), so there's no alternative implementation strategy. It's useless to require implementations to support driver dispatch for these cases.

I'm actually not sure that “symmetric keys” is the correct criterion. In theory, some asymmetric mechanisms could have a private key that are arbitrary byte strings, but none comes to mind right now. Curve25519 and Curve448 keys do come close: a few bit positions are forced to 0 or 1, but the rules are simple enough that there's no room for hardware acceleration. On the other side, DES keys aren't arbitrary byte strings, but the rules to calculate the forced bits are simple enough that no acceleration is relevant.